### PR TITLE
DOCS: API docs on Application and Collections

### PR DESCRIPTION
**Pull request by valadas**
## Summary
This adds/updates XML documentation on many methods and classes of the DotnetNuke.Application and DotNetNuke.Collections namespaces.

### Things needing help
I have not added any XML documentation on the following because I am no expert in that field (Locking and Shared collections). If anyone knows how to document that, either help me out here or on slack or by submitting an additional pull request, etc.

- [ ] DotNetNuke.Collections.ExclusiveLockStrategy
- [ ] DotNetNuke.Collections.ILockStrategy
- [ ] DotNetNuke.Collections.ISharedCollectionLock
- [ ] DotNetNuke.Collections.LockingStrategy
- [ ] LockingStrategyFactory
- [ ] MonitorLock
- [ ] NaiveLockingList
- [ ] ReaderWriterLockStrategy
- [ ] ReaderWriterSlimLock
- [ ] ReadLockRequiredException
- [ ] SharedDictionary
- [ ] SharedList
- [ ] WriteLockRequiredException

### CODE REVIEW

File: DNN Platform/Library/Collections/CollectionExtensions.cs [hunk 1]
```diff
@@ -34,12 +34,21 @@
...
     public static class CollectionExtensions
     {
         private static readonly ILog Logger = LoggerSource.Instance.GetLogger(typeof(CollectionExtensions));
+
         /// <summary>
-        /// This method is extracted from the FriendlyUrlSettings class
+        /// Converts a string with multiple key-value pairs into a Dictionary, if there are duplicated keys in your string 
+        /// only the first found key is kept since keys must be unique in a Dictionary
         /// </summary>
+        /// <param name="stringOfPairs">The string containaing the key-value pairs</param>
```
Comment (bdukes) [inline DNN Platform/Library/Collections/CollectionExtensions.cs]:
```suggestion
        /// <param name="stringOfPairs">The string containing the key-value pairs</param> 
```

File: DNN Platform/Library/Collections/CollectionExtensions.cs [hunk 2]
```diff
@@ -34,12 +34,21 @@
...
+        /// <param name="pairsSeparator">The separator used to separate each key-value pair</param>
+        /// <param name="pairSeparator">The separator used between the each key and value</param>
+        /// <returns>Dictionary with a string key and a sting value</returns>
```
Comment (bdukes) [inline DNN Platform/Library/Collections/CollectionExtensions.cs]:
```suggestion
        /// <returns>Dictionary with a string key and a string value</returns> 
```

File: DNN Platform/Library/Collections/CollectionExtensions.cs [hunk 3]
```diff
@@ -68,6 +77,13 @@ public static class CollectionExtensions
             return dictionary;
         }
 
+        /// <summary>
+        /// Converts a Dictionary with a string key and a string value into a plain string separated by your choice of separators
+        /// </summary>
+        /// <param name="dictionary">The Dictionary to convert</param>
+        /// <param name="pairsSeparator">The separator to use between key-value pairs</param>
+        /// <param name="pairSeparator">The separator to use between each key and value</param>
+        /// <returns></returns>
```
Comment (bdukes) [inline DNN Platform/Library/Collections/CollectionExtensions.cs]:
```suggestion
        /// <returns>The dictionary's contents as a <see cref="string" /></returns> 
```

File: DNN Platform/Library/Collections/PagedList.cs [hunk 4]
```diff
@@ -128,22 +128,22 @@ private void CommonConstruct(IEnumerable<T> items, int totalCount, int pageIndex
         public bool IsLastPage { get; private set; }
 
         /// <summary>
-        ///   The no of pages in this list
+        ///   Gets or sets the no of pages in this list
```
Comment (bdukes) [inline DNN Platform/Library/Collections/PagedList.cs]:
```suggestion
        ///   Gets or sets the number of pages in this list 
```

File: DNN Platform/Library/Collections/PagedList.cs [hunk 5]
```diff
@@ -128,22 +128,22 @@ private void CommonConstruct(IEnumerable<T> items, int totalCount, int pageIndex
...
 
         /// <summary>
-        ///   The index of the page contained in this list
+        ///   Gets or sets the index of the page contained in this list
         /// </summary>
         public int PageIndex { get; set; }
 
         /// <summary>
-        ///   The size of the page in this list
+        ///   Gets or sets he size of the page in this list
```
Comment (bdukes) [inline DNN Platform/Library/Collections/PagedList.cs]:
```suggestion
        ///   Gets or sets the size of the page in this list 
```

File: DNN Platform/Library/Collections/PagedList.cs [hunk 6]
```diff
@@ -128,22 +128,22 @@ private void CommonConstruct(IEnumerable<T> items, int totalCount, int pageIndex
...
         /// </summary>
         public int PageSize { get; set; }
 
         /// <summary>
-        ///   The total number of objects in the data store
+        ///   Gets or setshe total number of objects in the data store
```
Comment (bdukes) [inline DNN Platform/Library/Collections/PagedList.cs]:
```suggestion
        ///   Gets or ses the total number of objects in the data store 
```

File: DNN Platform/Library/Collections/PagedSelector.cs [hunk 7]
```diff
@@ -17,7 +17,7 @@ public class PageSelector<T>
         #region Constructors
 
         /// <summary>
-        /// Constructs a new PageSelector for use on the specified data store
+        /// Initializes a new instanc of the <see cref="PageSelector{T}"/> a new PageSelector for use on the specified data store
```
Comment (bdukes) [inline DNN Platform/Library/Collections/PagedSelector.cs]:
```suggestion
        /// Initializes a new instance of the <see cref="PageSelector{T}"/> for use on the specified data store 
```

File: DNN Platform/Library/Collections/PagingExtensions.cs [hunk 8]
```diff
@@ -52,12 +52,14 @@ public static PageSelector<T> InPagesOf<T>(this IEnumerable<T> source, int pageS
...
 
         /// <summary>
+        /// Converts an <see cref="IEnumerable{T}"/> into an <see cref="IPagedList{T}"/>
         /// </summary>
-        /// <typeparam name = "T"></typeparam>
-        /// <param name = "source"></param>
-        /// <param name = "pageIndex"></param>
-        /// <param name = "pageSize"></param>
-        /// <returns></returns>
+        /// <typeparam name = "T">The type of the <see cref="IEnumerable{T}"/></typeparam>
```
Comment (bdukes) [inline DNN Platform/Library/Collections/PagingExtensions.cs]:
```suggestion
        /// <typeparam name="T">The type of the items in the <paramref name="source"/></typeparam> 
```

**Comment by bdukes** [comment]
Thanks @valadas!

**Comment by valadas** [comment]
Thanks for the review @bdukes all accepted.

Depending on how you guys view this, either we coordinate to fill up all the checkboxes in my initial comment and we have completed those namespaces, or if you prefer to merge smaller pull requests, then you can merge this and we can revisit. I found out we can analyse file by file using Stylecop for missing XML docs, so it would be easy to stop what does not have it without raising the warning levels in Visual Studio.

**Comment by bdukes** [comment]
I'd lean towards merging what we have, and then moving the checklist into an issue

**Comment by valadas** [comment]
Makes sense

**Comment by valadas** [comment]
Moved the list of missing methods to document to #2429

**Pull request merged**